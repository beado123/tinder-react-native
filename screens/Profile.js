import React from 'react';
import styles from '../styles/Profile-style.js'
import { StyleSheet, Text, View, Image, TouchableOpacity, ImageBackground } from 'react-native';
import  {connect} from 'react-redux';
import Icon from 'react-native-vector-icons/FontAwesome';

import {uploadImage, logout} from '../redux/actions.js';

export class Profile extends React.Component {

    /*Sign out from app*/
    logout = () => {
        this.props.dispatch(logout());
        this.props.navigation.navigate('Main');
    }

    render() {
        return (
            <View style={{flex:1,flexDirection: 'column'}}>
                <View style={styles.profileUpper}>
                    <View style={styles.profilepicWrap}>
                        <Image style={styles.profilepic} source={{uri: this.props.user.photoUrl}} />
                    </View>
                    <Text style={styles.name}>{this.props.user.showAge === false? this.props.user.name: this.props.user.name+','+ (2018-Number(this.props.user.birth))}</Text>
                    <View  style={{flexDirection: 'row', marginTop: 110}}>
                        <Icon name='edit' size={25} style={styles.editIcon} />
                        <TouchableOpacity style = {styles.ProfileButtonLogin} onPress={() => this.props.navigation.navigate('Profile_Update')}>
                            <Text style={styles.ProfileButtonText} >Update Profile</Text>
                        </TouchableOpacity>
                    </View>

                </View>
                <View style={styles.profileLower}>
                    <View style={{flexDirection: 'row'}}>
                        <Icon name='sign-out' size={25} style={styles.signoutIcon} />
                        <TouchableOpacity style = {styles.ProfileButtonSignout} onPress = {() => this.logout()} >
                            <Text style={styles.ProfileButtonText} >SignOut</Text>
                        </TouchableOpacity>
                    </View>

                </View>
            </View>
        );
    }
}

/*Connect state from the store to props, state.user corresponds to action.user in reducer.js*/
function mapStateToProps(state) {
    console.log('loading profile page');
    return {
        user: state.user
    };
}

/*Connects a React component to a Redux store*/
export default connect(mapStateToProps)(Profile);
