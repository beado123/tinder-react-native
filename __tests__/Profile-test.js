import React from 'react';
import {Profile} from '../screens/Profile.js';
import renderer from 'react-test-renderer';
import "isomorphic-fetch"

test('renders correctly', () => {
  const tree = renderer.create(<Profile user={{photoUrl: 'https//mock_email', name: 'mock_name'}}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
