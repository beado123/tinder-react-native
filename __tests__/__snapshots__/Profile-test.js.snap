// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 3,
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https//mock_email",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 80,
            "justifyContent": "center",
            "marginTop": 40,
            "width": 80,
          }
        }
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "Georgia",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginTop": 30,
        }
      }
    >
      mock_name,NaN
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 110,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#f4f2f5",
            "borderRadius": 20,
            "marginTop": 10,
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontFamily": "Georgia",
              "fontSize": 14,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Update Profile
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f4f2f5",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 20,
            "opacity": 1,
            "padding": 15,
            "width": 100,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontFamily": "Georgia",
              "fontSize": 14,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          SignOut
        </Text>
      </View>
    </View>
  </View>
</View>
`;
